src/auro-input.js

```diff
@@ -106,16 +114,140 @@ export default class AuroInput extends BaseInput {
     return null
   }
 
+  connectedCallback() {
+    super.connectedCallback();
+
+    if (this.format) {
+      var cleave = new Cleave(this, this.format);
+    }
+  }
+
+  getCreditCardIcon() {
+    return this.icon && this.format && this.format.hasOwnProperty('creditCard') ?
+      html`
+        <div class="creditCard-icon">
+         ${this.generateCreditCardIconHtml()}
+        </div>
+      ` :
+      html``
+  }
+
+  generateCreditCardIconHtml() {
+    let icon; //
+
+    if (!this.value) {
+      icon = creditCard;
+    } else {
+      const v = this.value.trim();
+
+      if (v.length === 0) {
+        icon = creditCard;
+      }
+
+      if (v.length === 1) {
+        if (parseInt(v) === 4) {
+          // Visa
+          icon = creditCardVisa;
+        }
+      }
+
+      if (v.length === 2) {
+        const val = parseInt(v.substring(0, 2));
+
+        switch(val) {
+          case 34: case 37:
```

Comment:
For the future, it would be really handy to know where these magic numbers come from

Reply:
Good call out @gusnaughton - adding a link in a comment at the top of the method to https://www.creditcardinsider.com/learn/anatomy-of-a-credit-card/#credit-card-number